api/resolvers/growth.js

```diff
@@ -44,50 +57,63 @@ export function withClause (when) {
 }
 
 // HACKY AF this is a performance enhancement that allows us to use the created_at indices on tables
-export function intervalClause (when, table, and) {
+export function intervalClause (when, from, to, table, and) {
   const unit = timeUnit(when)
-  if (when === 'forever') {
-    return and ? '' : 'TRUE'
+  switch (when) {
+    case 'forever':
+      return and ? '' : 'TRUE'
+    case 'custom':
+      return `"${table}".created_at >= ${from} AND "${table}".created_at <= ${to} ${and ? 'AND' : ''} `
+    default:
+      return `"${table}".created_at >= date_trunc('${unit}', now_utc() - interval '${interval(when)}') ${and ? 'AND' : ''} `
   }
-
-  return `"${table}".created_at >= date_trunc('${unit}', now_utc() - interval '${interval(when)}') ${and ? 'AND' : ''} `
 }
 
-export function viewIntervalClause (when, view, and) {
-  if (when === 'forever') {
-    return and ? '' : 'TRUE'
+export function viewIntervalClause (when, from, to, view, and) {
+  function daysBetween (one, another) { // horrible hack for custom case
+    return Math.round(Math.abs((+new Date(one)) - (+new Date(another))) / 8.64e7)
+  }
+  switch (when) {
+    case 'forever':
+      return and ? '' : 'TRUE'
+    case 'custom':
+      return `"${view}".day >= date_trunc('day', timezone('America/Chicago', now() - interval '${daysBetween(new Date(), from) + ' days'}')) AND "${view}".day <= date_trunc('day', timezone('America/Chicago', now() - interval '${daysBetween(new Date(), to) + ' days'}')) ${and ? 'AND' : ''} `
```

Comment:
And this is stranger than strange. As written, it "works" but obviously its the hack of all hacks. I would simply like to put something such as `>= $1` (or `>= ${from}`), but when I do so, the results are as if the $1 parameter is always "now()" even when it isn't.

Reply:
This seems a little hairy. I won't be able to diagnose without pulling which I won't be able to do for another couple days.

Reply:
I couldn't get this to work as-is. Using `forever` shows this:

![2023-10-30-131208_1920x1080_scrot](https://github.com/stackernews/stacker.news/assets/27162016/164bfe1b-0bf5-4ffe-894b-9d1e60797148)

Using `custom` from `01/01/22` to `10/29/23`:

![2023-10-30-131232_1920x1080_scrot](https://github.com/stackernews/stacker.news/assets/27162016/dbb364ae-cb8c-4f60-a5cc-2e07aeb50b6e)


Shouldn't I get results with this query if it would work? :thinking: 

I wanted to investigate this:

> when I do so, the results are as if the $1 parameter is always "now" even when it isn't.

but since you mentioned, as it is, it works but it doesn't for me, I am not sure where to start.